setup.py

```diff
@@ -36,18 +37,56 @@ def run(self):
 cmdclass = versioneer.get_cmdclass()
 cmdclass['build_doc'] = build_doc
 
-
-GCCFLAGS = ["-std=c89", "-Wdeclaration-after-statement", "-Werror"]
-
-if os.environ.get("NUMBA_GCC_FLAGS"):
-    CFLAGS = GCCFLAGS
-else:
-    CFLAGS = ['-g']
-
 install_name_tool_fixer = []
 if sys.platform == 'darwin':
     install_name_tool_fixer += ['-headerpad_max_install_names']
 
+build_ext = cmdclass.get('build_ext', build_ext)
+
+numba_be_user_options = [
+    ('werror', None, 'Build extensions with -Werror'),
+    ('wall', None, 'Build extensions with -Wall'),
+    ('noopt', None, 'Build extensions without optimization'),
+]
+
+
+class NumbaBuildExt(build_ext):
+
+    user_options = build_ext.user_options + numba_be_user_options
+    boolean_options = build_ext.boolean_options + ['werror', 'wall', 'noopt']
+
+    def initialize_options(self):
+        super().initialize_options()
+        # We build with -Werror and -Wall by default on Linux x86, to provide
+        # some level of checking without the burden of maintaining it for all
+        # platforms.
+        is_linux_x86 = (sys.platform == 'linux' and platform.machine()
+                        in ('i386', 'x86_64'))
+        self.werror = is_linux_x86
+        self.wall = is_linux_x86
+        # Building with debug symbols should generally work on Linux and OS X.
+        self.debug = sys.platform in ('linux', 'darwin')
+        # Optional build without optimization, to make debugging extensions
+        # easier.
+        self.noopt = 0
```

Comment:
```suggestion
        self.noopt = False
```

Reply:
I would have done that, but I was following the convention in distutils command classes.

Reply:
ah ok, I guess leave as is :)